### PR TITLE
Use contentContainerStyle to adjust totalDimension

### PR DESCRIPTION
**Pull request by levic92**
When `maxWidth` or `maxHeight` is specified in `contentContainerStyle` use this to limit `totalDimension`
When any padding is specified in `contentContainerStyle` subtract the space from `totalDimension`

Logic is located in utils function `adjustDimension` + use this function to set `defaultTotalDimension`. Previous use of `maxDimension` prop is still in place, and functionality does not change unless `contentContainerStyle` (with monitored attributes) is used.

### CODE REVIEW

**Comment by levic92** [comment]
I have made one more change to this proposed PR - use the padding from the style prop to adjust the `totalDimension` as well

This would potentially influence the styling that people using older versions of the library are used to seeing. Please review the code when you have time and we can discuss further if you decide you'd like to incorporate these changes. No rush, I am using my own branch in my app but I'd like to contribute these changes as I believe they are helpful in keeping the styling of an app consistent.

I will explain the use case for introducing the adjustment based on the padding in the `style` prop:
I use the `contentContainerStyle` prop to keep the styling and padding inline with the regular scroll views I use in my app (ex: maxWidth 1000 with 20px horizontal padding). Now the reason for adding padding in the `style` prop as well is to match the safe area insets specifically on iOS. In horizontal orientation there is a 47px inset on either side and instead of using another component and insetting the actual ScrollView, I would rather add this extra 47px padding on either side of the scrollView so that it still is the full width of the screen for touches. I imagine there could be other use cases.

**Comment by levic92** [comment]
As mentioned in the other PR maybe add a config option like `adjustGridToStyles` that enables or disables this new behaviour

**Comment by levic92** [comment]
As mentioned in the other PR maybe add a config option like `adjustGridToStyles` that enables or disables this new behaviour

File: utils.js [hunk 1]
```diff
@@ -41,6 +42,83 @@ function calculateDimensions({
...
+    const padding = flatStyle[sPaddingXY] || flatStyle.padding;
+    const padding1 = flatStyle[sPadding1] || padding || 0;
+    const padding2 = flatStyle[sPadding2] || padding || 0;
+    space1 = (typeof padding1 === 'number' ? padding1 : 0);
+    space2 = (typeof padding2 === 'number' ? padding2 : 0);
+  }
+  return { space1, space2, maxStyleDimension };
+}
+
+function adjustDimension({
```
Comment (saleel) [inline utils.js:78]: I would call this `getStyleAdjustedTotalDimensions`, and the first parameter can `totalDimension` instead of `newTotalDimension`

**Review by saleel — APPROVED** [review]
Looks good 🚀 

Just left a small comment

**Comment by saleel** [comment]
This is amazing work 👍 

> As mentioned in the other PR maybe add a config option like `adjustGridToStyles` that enables or disables this new behavior

I would say we go this and have this change sit behind a boolean prop (we can document this properly).

**Comment by levic92** [comment]
Do we call this boolean prop `adjustGridToStyles` ?
Should it be enabled or disabled by default?

**Comment by saleel** [comment]
Yes, `adjustGridToStyles` sounds good. I would say disabled by default so existing behavior don't change for anyone.

**Comment by levic92** [comment]
@saleel the requested changes have been made

**Review by saleel — APPROVED** [review]

**Comment by saleel** [comment]
@levic92 Looks good. Thanks

I have merged and released 4.4.0. I have reverted the changes from last two PRs (as they are already covered in this one)

**Pull request merged**